botocore/session.py

```diff
@@ -779,7 +779,9 @@ def create_client(self, service_name, region_name=None, api_version=None,
         event_emitter = self.get_component('event_emitter')
         response_parser_factory = self.get_component(
             'response_parser_factory')
-        if aws_secret_access_key is not None:
+        if config and config.signature_version == 'unsigned':
```

Comment:
Can we also add a functional or unit test to make sure ``UNSIGNED`` is used for the signature if it is specified in the client config as we are now making it a full fledged feature?